lib/Doctrine/Common/Annotations/DocParser.php

```diff
@@ -823,7 +823,11 @@ private function Annotation()
         foreach ($values as $property => $value) {
             if (!isset(self::$annotationMetadata[$name]['properties'][$property])) {
                 if ('value' !== $property) {
-                    throw AnnotationException::creationError(sprintf('The annotation @%s declared on %s does not have a property named "%s". Available properties: %s', $originalName, $this->context, $property, implode(', ', self::$annotationMetadata[$name]['properties'])));
+                    if (self::$annotationMetadata[$name]['properties']) {
+                        throw AnnotationException::creationError(sprintf('The annotation @%s declared on %s does not have a property named "%s". Available properties: %s.', $originalName, $this->context, $property, implode(', ', self::$annotationMetadata[$name]['properties'])));
+                    } else {
```

Comment:
`else` is redundant - if it throws, no `else` is needed

Reply:
oh yes. Fixed